packages/widgets/src/blocks/legacy-widget/index.php

```diff
@@ -24,15 +24,38 @@ function render_block_core_legacy_widget( $attributes ) {
 		return '';
 	}
 
-	$id_base = $attributes['idBase'];
+	$id_base       = $attributes['idBase'];
+	$widget_key    = '';
+	$widget_object = null;
+
 	if ( method_exists( $wp_widget_factory, 'get_widget_key' ) && method_exists( $wp_widget_factory, 'get_widget_object' ) ) {
 		$widget_key    = $wp_widget_factory->get_widget_key( $id_base );
 		$widget_object = $wp_widget_factory->get_widget_object( $id_base );
-	} else {
-		$widget_key    = gutenberg_get_widget_key( $id_base );
-		$widget_object = gutenberg_get_widget_object( $id_base );
 	}
 
+	/**
+	 * Filters the core legacy widget block key.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param string $widget_key The widget key.
+	 * @param string $id_base    The widget ID base.
+	 * @param array  $attributes The block attributes.
+	 */
+	$widget_key = apply_filters( 'render_block_core_legacy_widget_key', $widget_key, $id_base, $attributes );
```

Comment:
Do we really want these filters in core or are they here just for the Gutenberg plugin and in which case we should mark them `__unstable` or internal or something.

Reply:
After discussion on the Trac ticket, the compromise of including a detailed inline comment about the two lines of code would be acceptable for this instance. Refreshing the PR now.